src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -170,7 +170,7 @@ module.exports = function<T, P, I, TI, PI, C, CX>(config : HostConfig<T, P, I, T
 
     deferredUpdates,
 
-    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | I | TI | null) {
+    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | I | TI | PI | null) {
```

Comment:
The purpose of this method is the same as for refs except for the return value on the very top level. `var inst = ReactDOM.render(...)`

It should never be possible to get a handle on an instance of `I` or `TI` since those are internal instances to the renderer. The point of `getPublicInstance` is to ensure that.

Therefore this should only return `ReactComponent<any, any, any> | PI | null`.

To ensure that you must call `getPublicInstance` on the `stateNode` for the appropriate tag.